### PR TITLE
compiler-rt: cont. remove <cyclades.h> from libsanitizer

### PR DESCRIPTION
**Pull request by Princemachiavelli**
From commit 199b7c505b0390429b08edf68552e8a6ff4a08ce for the
original issue:

>  linux-headers-5.13 removed <cyclades.h> along with device support.
    Backport a single https://reviews.llvm.org/D102059 upstream change to
    fix compiler-rt build.

This only unique thing here is the patch is from the upstream rocm llvm repo:
https://reviews.llvm.org/rG884040db086936107ec81656aa5b4c607235fb9a

With the patch, the  llvmPackages_rocm.compiler-rt package does compile. Once hydra shows the build failure I can open an issue if needed.
 
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Princemachiavelli** [comment]
Waiting to see what nixpkgs-review finds. I'm not quite sure what downstream packages have this as a build dependency.

File: pkgs/development/compilers/llvm/rocm/compiler-rt/default.nix [hunk 1]
```diff
@@ -31,7 +31,7 @@ stdenv.mkDerivation rec {
 
   patches = [
     ./compiler-rt-codesign.patch # Revert compiler-rt commit that makes codesign mandatory
-  ];
+  ] ++ [ ../../common/compiler-rt/libsanitizer-no-cyclades-rocm.patch ];
```
Comment (Mic92) [inline pkgs/development/compilers/llvm/rocm/compiler-rt/default.nix:34]: The patch is quite large. Did you create this patch yourself? If not can we use fetchpatch?
Otherwise it seems that the chunksize is quite big when patching because it contains a lot of unrelated lines.

Comment (Mic92) [reply pkgs/development/compilers/llvm/rocm/compiler-rt/default.nix:34]: After reading your comment, my question is answered:

This would be the preferred way:

```
(fetchpatch { url = "https://reviews.llvm.org/file/data/nrhbpc5axblqwx2xyyzv/PHID-FILE-wwcpjvquusomoddmqcwo/file"; sha256 = "<the actual hash>"; })
```

**Comment by acowley** [comment]
The `hip` package is a significant downstream dependent. I don’t think we have an automated test suite for it, but building it is good.

**Comment by Princemachiavelli** [comment]
> The `hip` package is a significant downstream dependent. I don’t think we have an automated test suite for it, but building it is good.

Thanks. I was able to build the `hip` package without issue.

**Review by acowley — APPROVED** [review]
Looks good! I'm glad you were able to `fetchpatch` from upstream. I built and lightly tested a `hip` with this patch and all is well.

**Comment by lovesegfault** [comment]
Result of `nixpkgs-review pr 139812` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>hip</li>
    <li>llvmPackages_rocm.clang</li>
    <li>llvmPackages_rocm.compiler-rt</li>
    <li>rocclr</li>
    <li>rocm-comgr</li>
    <li>rocm-device-libs</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>rocm-runtime</li>
    <li>rocminfo</li>
  </ul>
</details>

**Comment by lovesegfault** [comment]
This (sadly) broke things now due to the patch having gone 404, I'm working on a fix.
